code/Model/RedirectorPage.php

```diff
@@ -171,35 +175,12 @@ public function syncLinkTracking()
         }
     }
 
-    protected function onBeforeWrite()
-    {
-        parent::onBeforeWrite();
-
-        if ($this->ExternalURL && substr($this->ExternalURL ?? '', 0, 2) !== '//') {
-            $urlParts = parse_url($this->ExternalURL ?? '');
-            if ($urlParts) {
-                if (empty($urlParts['scheme'])) {
-                    // no scheme, assume http
-                    $this->ExternalURL = 'http://' . $this->ExternalURL;
-                } elseif (!in_array($urlParts['scheme'], [
-                    'http',
-                    'https',
-                ])) {
-                    // we only allow http(s) urls
-                    $this->ExternalURL = '';
-                }
-            } else {
-                // malformed URL to reject
-                $this->ExternalURL = '';
-            }
-        }
-    }
```

Comment:
No need to set false if it's invalid, because the `URLField` validation won't allow an invalid value in the first place.
The trade off is we're not assuming `http` by default anymore, users will have to add that themselves.
The `URLField` validation has been updated to make it clear when that needs to be added.